pkgs/development/python-modules/textx/default.nix

```
@@ -0,0 +1,69 @@
+{ lib
+, buildPythonPackage
+, python3
+, fetchFromGitHub
+, mkdocs
+, twine
+, arpeggio
+, click
+, future
+, coverage
+, coveralls
+, flake8
+, html5lib
+, jinja2
+, pytest
+, tox
+}:
+
+buildPythonPackage rec {
+  pname = "textx";
+  version = "3.0.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-uZlO82dKtWQQR5+Q7dWk3+ZoUzAjDJ8qzC4UMLCtnBk=";
+  };
+
+  nativeBuildInputs = [
+    mkdocs
+    twine
+  ];
+
+  propagatedBuildInputs = [
+    arpeggio
+    click
+    future
+  ];
+
+  checkInputs = [
+    click
+    coverage
+    coveralls
+    flake8
+    html5lib
+    jinja2
+    pytest
+    tox
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.cfg --replace "click >=7.0, <8.0" "click >=7.0"
+  '';
+
+  # Many import errors.
+  doCheck = false;
```

For missing other packages or self references?

---

```sh
python3.9-textx> no Makefile or custom installCheckPhase, doing nothing
python3.9-textx> pythonCatchConflictsPhase
python3.9-textx> pythonRemoveBinBytecodePhase
python3.9-textx> pythonImportsCheckPhase
python3.9-textx> Executing pythonImportsCheckPhase
python3.9-textx> Check whether the following modules can be imported: textx
python3.9-textx> setuptoolsCheckPhase
python3.9-textx> Executing setuptoolsCheckPhase
python3.9-textx> running test
python3.9-textx> WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
python3.9-textx> running egg_info
python3.9-textx> writing textX.egg-info/PKG-INFO
python3.9-textx> writing dependency_links to textX.egg-info/dependency_links.txt
python3.9-textx> writing entry points to textX.egg-info/entry_points.txt
python3.9-textx> writing requirements to textX.egg-info/requires.txt
python3.9-textx> writing top-level names to textX.egg-info/top_level.txt
python3.9-textx> reading manifest file 'textX.egg-info/SOURCES.txt'
python3.9-textx> adding license file 'LICENSE.txt'
python3.9-textx> adding license file 'AUTHORS.md'
python3.9-textx> running build_ext
python3.9-textx> six (unittest.loader._FailedTest) ... ERROR
python3.9-textx> const (unittest.loader._FailedTest) ... ERROR
python3.9-textx> generators (unittest.loader._FailedTest) ... ERROR
python3.9-textx> model_params (unittest.loader._FailedTest) ... ERROR
python3.9-textx> registration (unittest.loader._FailedTest) ... ERROR
python3.9-textx> cli (unittest.loader._FailedTest) ... ERROR
python3.9-textx> exceptions (unittest.loader._FailedTest) ... ERROR
python3.9-textx> export (unittest.loader._FailedTest) ... ERROR
python3.9-textx> lang (unittest.loader._FailedTest) ... ERROR
python3.9-textx> metamodel (unittest.loader._FailedTest) ... ERROR
python3.9-textx> model (unittest.loader._FailedTest) ... ERROR
python3.9-textx> scoping (unittest.loader._FailedTest) ... ERROR
python3.9-textx> ======================================================================
python3.9-textx> ERROR: six (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: six
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.six'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: const (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: const
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.const'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: generators (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: generators
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.generators'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: model_params (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: model_params
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.model_params'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: registration (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: registration
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.registration'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: cli (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: cli
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.cli'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: exceptions (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: exceptions
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.exceptions'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: export (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: export
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.export'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: lang (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: lang
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.lang'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: metamodel (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: metamodel
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.metamodel'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: model (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: model
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.model'
python3.9-textx> ======================================================================
python3.9-textx> ERROR: scoping (unittest.loader._FailedTest)
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> ImportError: Failed to import test module: scoping
python3.9-textx> Traceback (most recent call last):
python3.9-textx>   File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/unittest/loader.py", line 154, in loadTestsFromName
python3.9-textx>     module = __import__(module_name)
python3.9-textx> ModuleNotFoundError: No module named 'textx.six.scoping'
python3.9-textx> ----------------------------------------------------------------------
python3.9-textx> Ran 12 tests in 0.003s
python3.9-textx> FAILED (errors=12)
python3.9-textx> Test failed: <unittest.runner.TextTestResult run=12 errors=12 failures=0>
python3.9-textx> error: Test failed: <unittest.runner.TextTestResult run=12 errors=12 failures=0>
```

---

Can you try executing the tests with pytest or whatever software upstream is using? If that does not help please add a comment like:

```suggestion
  # fails with many similar errors like:
  # ModuleNotFoundError: No module named 'textx.six.scoping'
  doCheck = false;
```

---

i added gprof2dot, memory_profile, psutil and enabled pytestcheckhook.

now i get different errors, some seem due to hardcoded paths... i'll need to look at it more

```
python3.9-textx> running install tests
python3.9-textx> no Makefile or custom installCheckPhase, doing nothing
python3.9-textx> pythonCatchConflictsPhase
python3.9-textx> pythonRemoveBinBytecodePhase
python3.9-textx> pythonImportsCheckPhase
python3.9-textx> Executing pythonImportsCheckPhase
python3.9-textx> Check whether the following modules can be imported: textx
python3.9-textx> pytestCheckPhase
python3.9-textx> Executing pytestCheckPhase
python3.9-textx> ============================= test session starts ==============================
python3.9-textx> platform linux -- Python 3.9.11, pytest-7.0.1, pluggy-1.0.0
python3.9-textx> rootdir: /build/source
python3.9-textx> collected 331 items
python3.9-textx> tests/functional/test_auto_init.py ..                                    [  0%]
python3.9-textx> tests/functional/test_examples.py .                                      [  0%]
python3.9-textx> tests/functional/test_metaclass_reference.py .....                       [  2%]
python3.9-textx> tests/functional/test_metamodel_params.py ....                           [  3%]
python3.9-textx> tests/functional/test_metamodel_properties.py ...                        [  4%]
python3.9-textx> tests/functional/test_model_api.py ....                                  [  5%]
python3.9-textx> tests/functional/test_objcrossref_positions.py .                         [  6%]
python3.9-textx> tests/functional/test_processors.py ..............                       [ 10%]
python3.9-textx> tests/functional/test_regexp_with_groups.py ...                          [ 11%]
python3.9-textx> tests/functional/test_repeat_modifiers.py ............                   [ 14%]
python3.9-textx> tests/functional/test_resolving.py ..                                    [ 15%]
python3.9-textx> tests/functional/test_rule_modifiers.py ....                             [ 16%]
python3.9-textx> tests/functional/test_special_model_attributes.py .                      [ 16%]
python3.9-textx> tests/functional/test_strict_unicode.py .                                [ 17%]
python3.9-textx> tests/functional/test_textx_language.py ................................ [ 26%]
python3.9-textx> ..........                                                               [ 29%]
python3.9-textx> tests/functional/test_textx_tools_support.py .                           [ 30%]
python3.9-textx> tests/functional/test_unicode.py .                                       [ 30%]
python3.9-textx> tests/functional/test_user_class_attrs_frozen.py ...                     [ 31%]
python3.9-textx> tests/functional/test_user_class_constructor_calls.py ...                [ 32%]
python3.9-textx> tests/functional/test_user_class_with_slots.py .                         [ 32%]
python3.9-textx> tests/functional/test_user_classes.py ..                                 [ 33%]
python3.9-textx> tests/functional/test_user_classes_attr_functions.py .                   [ 33%]
python3.9-textx> tests/functional/test_user_classes_callable.py .                         [ 33%]
python3.9-textx> tests/functional/test_user_classes_init_order.py .                       [ 34%]
python3.9-textx> tests/functional/test_user_classes_with_properties.py .                  [ 34%]
python3.9-textx> tests/functional/examples/test_free_text_with_references.py .            [ 34%]
python3.9-textx> tests/functional/examples/test_hierarchical_data_structures_referencing_attributes.py . [ 35%]
python3.9-textx> ...                                                                      [ 35%]
python3.9-textx> tests/functional/examples/test_modeling_float_int_variables.py .         [ 36%]
python3.9-textx> tests/functional/examples/test_reference_to_python.py .                  [ 36%]
python3.9-textx> tests/functional/generators/test_textx_generators.py ...                 [ 37%]
python3.9-textx> tests/functional/registration/test_check.py FFFFFF.F.                    [ 40%]
python3.9-textx> tests/functional/registration/test_discovery.py FF                       [ 40%]
python3.9-textx> tests/functional/registration/test_generate.py FFFF.                     [ 42%]
python3.9-textx> tests/functional/registration/test_registration.py .................F    [ 47%]
python3.9-textx> tests/functional/registration/test_version.py .                          [ 48%]
python3.9-textx> tests/functional/registration/projects/data_dsl/tests/test_data_dsl.py F [ 48%]
python3.9-textx>                                                                          [ 48%]
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py F [ 48%]
python3.9-textx> FF                                                                       [ 49%]
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_issue196.py F [ 49%]
python3.9-textx> FF                                                                       [ 50%]
python3.9-textx> tests/functional/registration/projects/types_dsl/tests/test_types_dsl.py F [ 50%]
python3.9-textx> F                                                                        [ 50%]
python3.9-textx> tests/functional/regressions/test_issue103_python_like_import.py ..      [ 51%]
python3.9-textx> tests/functional/regressions/test_issue108_obj_proc_multifile.py .       [ 51%]
python3.9-textx> tests/functional/regressions/test_issue128.py ..                         [ 52%]
python3.9-textx> tests/functional/regressions/test_issue140.py .                          [ 52%]
python3.9-textx> tests/functional/regressions/test_issue14_obj_proc.py .                  [ 52%]
python3.9-textx> tests/functional/regressions/test_issue155.py .                          [ 53%]
python3.9-textx> tests/functional/regressions/test_issue166.py .                          [ 53%]
python3.9-textx> tests/functional/regressions/test_issue188.py .....                      [ 54%]
python3.9-textx> tests/functional/regressions/test_issue205_skipws_propagation.py .       [ 55%]
python3.9-textx> tests/functional/regressions/test_issue206_regex.py .                    [ 55%]
python3.9-textx> tests/functional/regressions/test_issue23_direct_recursion.py .          [ 55%]
python3.9-textx> tests/functional/regressions/test_issue246.py .                          [ 56%]
python3.9-textx> tests/functional/regressions/test_issue275.py .                          [ 56%]
python3.9-textx> tests/functional/regressions/test_issue310.py .                          [ 56%]
python3.9-textx> tests/functional/regressions/test_issue33.py ...                         [ 57%]
python3.9-textx> tests/functional/regressions/test_issue358.py .......                    [ 59%]
python3.9-textx> tests/functional/regressions/test_issue360.py .                          [ 60%]
python3.9-textx> tests/functional/regressions/test_issue72.py .                           [ 60%]
python3.9-textx> tests/functional/regressions/test_issue78.py ......                      [ 62%]
python3.9-textx> tests/functional/regressions/test_issue96.py ..                          [ 62%]
python3.9-textx> tests/functional/regressions/test_issue97.py ..                          [ 63%]
python3.9-textx> tests/functional/regressions/test_issue_34_resolving.py .                [ 63%]
python3.9-textx> tests/functional/regressions/test_issue_53_comments.py .                 [ 64%]
python3.9-textx> tests/functional/regressions/test_issue_80_object_processors.py .        [ 64%]
python3.9-textx> tests/functional/regressions/test_issue_89.py .                          [ 64%]
python3.9-textx> tests/functional/regressions/test_json_issue.py .                        [ 64%]
python3.9-textx> tests/functional/regressions/test_repetition_single_assignment_error.py . [ 65%]
python3.9-textx> .                                                                        [ 65%]
python3.9-textx> tests/functional/regressions/issue323/test_issue323_backslash_chars.py . [ 65%]
python3.9-textx> .                                                                        [ 66%]
python3.9-textx> tests/functional/subcommands/test_subcommands.py FF                      [ 66%]
python3.9-textx> tests/functional/test_import/test_import.py ...                          [ 67%]
python3.9-textx> tests/functional/test_language_reference/test_language_reference.py .    [ 67%]
python3.9-textx> tests/functional/test_metamodel/test_metamodel.py .....                  [ 69%]
python3.9-textx> tests/functional/test_metamodel/test_model_params.py ....                [ 70%]
python3.9-textx> tests/functional/test_metamodel/test_multi_metamodel_refs.py .......     [ 72%]
python3.9-textx> tests/functional/test_scoping/test_buildins.py ..                        [ 73%]
python3.9-textx> tests/functional/test_scoping/test_builtin_models.py .                   [ 73%]
python3.9-textx> tests/functional/test_scoping/test_children.py .                         [ 74%]
python3.9-textx> tests/functional/test_scoping/test_exception_from_included_model.py .    [ 74%]
python3.9-textx> tests/functional/test_scoping/test_full_qualified_name.py ..             [ 74%]
python3.9-textx> tests/functional/test_scoping/test_full_qualified_name_rrel_manual.py .. [ 75%]
python3.9-textx> ...                                                                      [ 76%]
python3.9-textx> tests/functional/test_scoping/test_global_import_modules.py ....         [ 77%]
python3.9-textx> tests/functional/test_scoping/test_import_module.py ......               [ 79%]
python3.9-textx> tests/functional/test_scoping/test_import_module_search_path_issue66.py . [ 79%]
python3.9-textx> ......                                                                   [ 81%]
python3.9-textx> tests/functional/test_scoping/test_inheritance.py ..                     [ 82%]
python3.9-textx> tests/functional/test_scoping/test_issue170_combined_scope_provider.py . [ 82%]
python3.9-textx>                                                                          [ 82%]
python3.9-textx> tests/functional/test_scoping/test_local_scope.py ........               [ 84%]
python3.9-textx> tests/functional/test_scoping/test_local_scope_circular.py .             [ 85%]
python3.9-textx> tests/functional/test_scoping/test_metamodel_provider.py .               [ 85%]
python3.9-textx> tests/functional/test_scoping/test_metamodel_provider2.py .              [ 85%]
python3.9-textx> tests/functional/test_scoping/test_metamodel_provider3.py .....          [ 87%]
python3.9-textx> tests/functional/test_scoping/test_metamodel_provider3_custom_classes.py . [ 87%]
python3.9-textx> ....                                                                     [ 88%]
python3.9-textx> tests/functional/test_scoping/test_metamodel_provider_utf_16_le.py .     [ 89%]
python3.9-textx> tests/functional/test_scoping/test_model_export.py .                     [ 89%]
python3.9-textx> tests/functional/test_scoping/test_model_modification_through_scoping.py . [ 89%]
python3.9-textx> ....                                                                     [ 90%]
python3.9-textx> tests/functional/test_scoping/test_model_repository.py ...               [ 91%]
python3.9-textx> tests/functional/test_scoping/test_plain_name.py ...                     [ 92%]
python3.9-textx> tests/functional/test_scoping/test_reference_to_buildin_attribute.py .   [ 93%]
python3.9-textx> tests/functional/test_scoping/test_reference_to_nontextx_attribute.py .  [ 93%]
python3.9-textx> tests/functional/test_scoping/test_rrel.py ........                      [ 95%]
python3.9-textx> tests/functional/test_scoping/test_scoping_tools.py .......              [ 97%]
python3.9-textx> tests/functional/test_scoping/importURI_variations/test_importURI_variations.py . [ 98%]
python3.9-textx> ....                                                                     [ 99%]
python3.9-textx> tests/functional/textx_textx/test_textx_rrel.py .                        [ 99%]
python3.9-textx> tests/functional/textx_textx/test_textx_textx.py .                       [100%]
python3.9-textx> =================================== FAILURES ===================================
python3.9-textx> ____________ test_object_processor_with_optional_parameter_default _____________
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'types-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_object_processor_with_optional_parameter_default():
python3.9-textx> >       mm = metamodel_for_language('types-dsl')
python3.9-textx> tests/functional/registration/test_check.py:16:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "types-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> _______________ test_object_processor_with_optional_parameter_on _______________
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'types-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_object_processor_with_optional_parameter_on():
python3.9-textx> >       mm = metamodel_for_language('types-dsl')
python3.9-textx> tests/functional/registration/test_check.py:25:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "types-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ______________ test_object_processor_with_optional_parameter_off _______________
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'types-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_object_processor_with_optional_parameter_off():
python3.9-textx> >       mm = metamodel_for_language('types-dsl')
python3.9-textx> tests/functional/registration/test_check.py:34:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "types-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> _____________________________ test_check_metamodel _____________________________
python3.9-textx>     def test_check_metamodel():
python3.9-textx>         """
python3.9-textx>         Meta-model is also a model.
python3.9-textx>         """
python3.9-textx>         metamodel_file = os.path.join(this_folder,
python3.9-textx>                                       'projects', 'flow_dsl', 'flow_dsl',
python3.9-textx>                                       'Flow.tx')
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['check', metamodel_file])
python3.9-textx> >       assert result.exit_code == 0
python3.9-textx> E       assert 1 == 0
python3.9-textx> E        +  where 1 = <Result SystemExit(1)>.exit_code
python3.9-textx> tests/functional/registration/test_check.py:51: AssertionError
python3.9-textx> ____________________________ test_check_valid_model ____________________________
python3.9-textx>     def test_check_valid_model():
python3.9-textx>         model_file = os.path.join(this_folder,
python3.9-textx>                                   'projects', 'flow_dsl', 'tests',
python3.9-textx>                                   'models', 'data_flow.eflow')
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['check', model_file])
python3.9-textx> >       assert result.exit_code == 0
python3.9-textx> E       assert 1 == 0
python3.9-textx> E        +  where 1 = <Result SystemExit(1)>.exit_code
python3.9-textx> tests/functional/registration/test_check.py:61: AssertionError
python3.9-textx> ______________ test_check_valid_model_with_explicit_language_name ______________
python3.9-textx>     def test_check_valid_model_with_explicit_language_name():
python3.9-textx>         """
python3.9-textx>         Test checking of model where language name is given explicitly.
python3.9-textx>         """
python3.9-textx>         model_file = os.path.join(this_folder,
python3.9-textx>                                   'projects', 'flow_dsl', 'tests',
python3.9-textx>                                   'models', 'data_flow.eflow')
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['check',
python3.9-textx>                                        '--language', 'flow-dsl',
python3.9-textx>                                        model_file])
python3.9-textx> >       assert result.exit_code == 0
python3.9-textx> E       assert 1 == 0
python3.9-textx> E        +  where 1 = <Result SystemExit(1)>.exit_code
python3.9-textx> tests/functional/registration/test_check.py:76: AssertionError
python3.9-textx> ___________________________ test_check_invalid_model ___________________________
python3.9-textx>     def test_check_invalid_model():
python3.9-textx>         model_file = os.path.join(this_folder,
python3.9-textx>                                   'projects', 'flow_dsl', 'tests',
python3.9-textx>                                   'models', 'data_flow_including_error.eflow')
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['check', model_file])
python3.9-textx>         assert result.exit_code != 0
python3.9-textx>         assert 'Error:' in result.output
python3.9-textx> >       assert 'types must be lowercase' in result.output
python3.9-textx> E       assert 'types must be lowercase' in 'Error: No language registered that can parse "/build/source/tests/functional/registration/projects/flow_dsl/tests/models/data_flow_including_error.eflow".\n'
python3.9-textx> E        +  where 'Error: No language registered that can parse "/build/source/tests/functional/registration/projects/flow_dsl/tests/models/data_flow_including_error.eflow".\n' = <Result SystemExit(1)>.output
python3.9-textx> tests/functional/registration/test_check.py:106: AssertionError
python3.9-textx> ___________________________ test_list_languages_cli ____________________________
python3.9-textx>     def test_list_languages_cli():
python3.9-textx>         """
python3.9-textx>         Test list-languages command.
python3.9-textx>         """
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['list-languages'])
python3.9-textx>         assert result.exit_code == 0
python3.9-textx> >       assert 'flow-dsl[1.0.0]' in result.output
python3.9-textx> E       AssertionError: assert 'flow-dsl[1.0.0]' in 'textX (*.tx)                  textX[3.0.0]                            A meta-language for language definition\n'
python3.9-textx> E        +  where 'textX (*.tx)                  textX[3.0.0]                            A meta-language for language definition\n' = <Result okay>.output
python3.9-textx> tests/functional/registration/test_discovery.py:15: AssertionError
python3.9-textx> ___________________________ test_list_generators_cli ___________________________
python3.9-textx>     def test_list_generators_cli():
python3.9-textx>         """
python3.9-textx>         Test list-generators command.
python3.9-textx>         """
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['list-generators'])
python3.9-textx>         assert result.exit_code == 0
python3.9-textx> >       assert 'flow-dsl -> PlantUML' in result.output
python3.9-textx> E       AssertionError: assert 'flow-dsl -> PlantUML' in 'any -> dot                    textX[3.0.0]                  Generating dot visualizations from arbitrary models\ntext...\ntextX -> PlantUML             textX[3.0.0]                  Generating PlantUML visualizations from textX grammars\n'
python3.9-textx> E        +  where 'any -> dot                    textX[3.0.0]                  Generating dot visualizations from arbitrary models\ntext...\ntextX -> PlantUML             textX[3.0.0]                  Generating PlantUML visualizations from textX grammars\n' = <Result okay>.output
python3.9-textx> tests/functional/registration/test_discovery.py:27: AssertionError
python3.9-textx> __________________________ test_generator_registered ___________________________
python3.9-textx>     def test_generator_registered():
python3.9-textx>         """
python3.9-textx>         That that generator from flow to PlantUML is registered
python3.9-textx>         """
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['list-generators'])
python3.9-textx>         assert result.exit_code == 0
python3.9-textx> >       assert 'flow-dsl -> PlantUML' in result.output
python3.9-textx> E       AssertionError: assert 'flow-dsl -> PlantUML' in 'any -> dot                    textX[3.0.0]                  Generating dot visualizations from arbitrary models\ntext...\ntextX -> PlantUML             textX[3.0.0]                  Generating PlantUML visualizations from textX grammars\n'
python3.9-textx> E        +  where 'any -> dot                    textX[3.0.0]                  Generating dot visualizations from arbitrary models\ntext...\ntextX -> PlantUML             textX[3.0.0]                  Generating PlantUML visualizations from textX grammars\n' = <Result okay>.output
python3.9-textx> tests/functional/registration/test_generate.py:34: AssertionError
python3.9-textx> __________________________ test_generating_flow_model __________________________
python3.9-textx> model_file = '/build/source/tests/functional/registration/projects/flow_dsl/tests/models/data_flow.eflow'
python3.9-textx>     def test_generating_flow_model(model_file):
python3.9-textx>         """
python3.9-textx>         Test that generator can be called.
python3.9-textx>         """
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['generate',
python3.9-textx>                                        '--target', 'PlantUML',
python3.9-textx>                                        '--overwrite', model_file])
python3.9-textx> >       assert result.exit_code == 0
python3.9-textx> E       assert 1 == 0
python3.9-textx> E        +  where 1 = <Result SystemExit(1)>.exit_code
python3.9-textx> tests/functional/registration/test_generate.py:45: AssertionError
python3.9-textx> ______________ test_generate_by_providing_explicit_language_name _______________
python3.9-textx> model_file = '/build/source/tests/functional/registration/projects/flow_dsl/tests/models/data_flow.eflow'
python3.9-textx>     def test_generate_by_providing_explicit_language_name(model_file):
python3.9-textx>         """
python3.9-textx>         Test running generator by providing an explicit language name.
python3.9-textx>         """
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['generate',
python3.9-textx>                                        '--language', 'flow-dsl',
python3.9-textx>                                        '--target', 'PlantUML',
python3.9-textx>                                        '--overwrite', model_file])
python3.9-textx> >       assert result.exit_code == 0
python3.9-textx> E       assert 1 == 0
python3.9-textx> E        +  where 1 = <Result SystemExit(1)>.exit_code
python3.9-textx> tests/functional/registration/test_generate.py:63: AssertionError
python3.9-textx> __________________ test_passing_custom_arguments_to_generator __________________
python3.9-textx> model_file = '/build/source/tests/functional/registration/projects/flow_dsl/tests/models/data_flow.eflow'
python3.9-textx>     def test_passing_custom_arguments_to_generator(model_file):
python3.9-textx>         """
python3.9-textx>         Test passing custom arguments from command line to the generator.
python3.9-textx>         """
python3.9-textx>         runner = CliRunner()
python3.9-textx>         result = runner.invoke(textx, ['generate',
python3.9-textx>                                        '--language', 'flow-dsl',
python3.9-textx>                                        '--target', 'PlantUML',
python3.9-textx>                                        model_file,
python3.9-textx>                                        '--custom1', '42',
python3.9-textx>                                        '--custom2', '"some string"',
python3.9-textx>                                        '--overwrite'])
python3.9-textx> >       assert result.exit_code == 0
python3.9-textx> E       assert 1 == 0
python3.9-textx> E        +  where 1 = <Result SystemExit(1)>.exit_code
python3.9-textx> tests/functional/registration/test_generate.py:84: AssertionError
python3.9-textx> ___________ test_declaratively_registered_generator_always_available ___________
python3.9-textx> language_name = 'flow-dsl', target_name = 'plantuml', any_permitted = False
python3.9-textx>     def generator_description(language_name, target_name, any_permitted=False):
python3.9-textx>         """
python3.9-textx>         Return `GeneratorDesc` instance for the given target and language name.
python3.9-textx>         If `any_permitted` is `True` return generator for language `any` if
python3.9-textx>         generator for the `language_name` doesn't exist.
python3.9-textx>         """
python3.9-textx>         global generators
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         target_name = target_name.lower()
python3.9-textx>         if generators is None:
python3.9-textx>             generator_descriptions()
python3.9-textx>         try:
python3.9-textx>             try:
python3.9-textx> >               generators_for_language = generators[language_name]
python3.9-textx> E               KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:131: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_declaratively_registered_generator_always_available():
python3.9-textx>         """
python3.9-textx>         Declaratively registered generators will be re-registered at the first API
python3.9-textx>         call.
python3.9-textx>         """
python3.9-textx>         clear_generator_registrations()
python3.9-textx> >       generator = generator_description('flow-dsl', 'plantuml')
python3.9-textx> tests/functional/registration/test_registration.py:343:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl', target_name = 'plantuml', any_permitted = False
python3.9-textx>     def generator_description(language_name, target_name, any_permitted=False):
python3.9-textx>         """
python3.9-textx>         Return `GeneratorDesc` instance for the given target and language name.
python3.9-textx>         If `any_permitted` is `True` return generator for language `any` if
python3.9-textx>         generator for the `language_name` doesn't exist.
python3.9-textx>         """
python3.9-textx>         global generators
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         target_name = target_name.lower()
python3.9-textx>         if generators is None:
python3.9-textx>             generator_descriptions()
python3.9-textx>         try:
python3.9-textx>             try:
python3.9-textx>                 generators_for_language = generators[language_name]
python3.9-textx>                 return generators_for_language[target_name]
python3.9-textx>             except KeyError:
python3.9-textx>                 if not any_permitted:
python3.9-textx>                     raise
python3.9-textx>                 generators_for_language = generators['any']
python3.9-textx>                 return generators_for_language[target_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError(
python3.9-textx>                 'No generators registered for language "{}" and target "{}".'
python3.9-textx>                 .format(language_name, target_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'No generators registered for language "flow-dsl" and target "plantuml".'
python3.9-textx> textx/registration.py:139: TextXRegistrationError
python3.9-textx> ________________________________ test_data_dsl _________________________________
python3.9-textx> language_name = 'data-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'data-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_data_dsl():
python3.9-textx>         """
python3.9-textx>         Test loading of correct data dsl.
python3.9-textx>         """
python3.9-textx>         clear_language_registrations()
python3.9-textx>         current_dir = os.path.dirname(__file__)
python3.9-textx> >       mmD = metamodel_for_language('data-dsl')
python3.9-textx> tests/functional/registration/projects/data_dsl/tests/test_data_dsl.py:13:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'data-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "data-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ________________________________ test_flow_dsl _________________________________
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx> clear_all = None
python3.9-textx>     def test_flow_dsl(clear_all):
python3.9-textx>         """
python3.9-textx>         Test loading of correct flow model.
python3.9-textx>         """
python3.9-textx> >       mmF = metamodel_for_language('flow-dsl')
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py:22:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "flow-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ___________________________ test_flow_dsl_validation ___________________________
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx> clear_all = None
python3.9-textx>     def test_flow_dsl_validation(clear_all):
python3.9-textx>         """
python3.9-textx>         Test flow model with semantic error raises error.
python3.9-textx>         """
python3.9-textx> >       mmF = metamodel_for_language('flow-dsl')
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py:36:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "flow-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ________________________ test_flow_dsl_types_validation ________________________
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx> clear_all = None
python3.9-textx>     def test_flow_dsl_types_validation(clear_all):
python3.9-textx>         """
python3.9-textx>         Test flow model including error raises error.
python3.9-textx>         """
python3.9-textx>         # print("-----------------------------------1---")
python3.9-textx>         # print(metamodel_for_language('flow-dsl')._tx_model_repository.all_models.filename_to_model.keys())
python3.9-textx>         # print("----------------- #={}".format(
python3.9-textx>         #    len(metamodel_for_language('flow-dsl')._tx_model_repository.all_models.filename_to_model.keys())))
python3.9-textx> >       mmF = metamodel_for_language('flow-dsl')
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py:53:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "flow-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ___________ test_issue196_errors_in_scope_provider_and_obj_processor ___________
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_issue196_errors_in_scope_provider_and_obj_processor():
python3.9-textx> >       mmF = metamodel_for_language('flow-dsl')
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_issue196.py:11:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "flow-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> _________________________ test_issue196_syntax_error_1 _________________________
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_issue196_syntax_error_1():
python3.9-textx> >       mmF = metamodel_for_language('flow-dsl')
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_issue196.py:61:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "flow-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> _________________________ test_issue196_syntax_error_2 _________________________
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'flow-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx>     def test_issue196_syntax_error_2():
python3.9-textx> >       mmF = metamodel_for_language('flow-dsl')
python3.9-textx> tests/functional/registration/projects/flow_dsl/tests/test_issue196.py:87:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'flow-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "flow-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ________________________________ test_types_dsl ________________________________
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'types-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx> clear_all = None
python3.9-textx>     def test_types_dsl(clear_all):
python3.9-textx>         """
python3.9-textx>         Test loading of correct types model.
python3.9-textx>         """
python3.9-textx> >       mmT = metamodel_for_language('types-dsl')
python3.9-textx> tests/functional/registration/projects/types_dsl/tests/test_types_dsl.py:21:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "types-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> ____________________________ test_types_dsl_invalid ____________________________
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx> >           return languages[language_name]
python3.9-textx> E           KeyError: 'types-dsl'
python3.9-textx> textx/registration.py:112: KeyError
python3.9-textx> During handling of the above exception, another exception occurred:
python3.9-textx> clear_all = None
python3.9-textx>     def test_types_dsl_invalid(clear_all):
python3.9-textx>         """
python3.9-textx>         Test that types model with semantic error raises the error.
python3.9-textx>         """
python3.9-textx> >       mmT = metamodel_for_language('types-dsl')
python3.9-textx> tests/functional/registration/projects/types_dsl/tests/test_types_dsl.py:34:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> textx/registration.py:261: in metamodel_for_language
python3.9-textx>     language = language_description(language_name)
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> language_name = 'types-dsl'
python3.9-textx>     def language_description(language_name):
python3.9-textx>         """
python3.9-textx>         Return `LanguageDesc` for the given language name.
python3.9-textx>         """
python3.9-textx>         global languages
python3.9-textx>         language_name = language_name.lower()
python3.9-textx>         if languages is None:
python3.9-textx>             language_descriptions()
python3.9-textx>         try:
python3.9-textx>             return languages[language_name]
python3.9-textx>         except KeyError:
python3.9-textx> >           raise TextXRegistrationError('Language "{}" not registered.'
python3.9-textx>                                          .format(language_name))
python3.9-textx> E           textx.exceptions.TextXRegistrationError: b'Language "types-dsl" not registered.'
python3.9-textx> textx/registration.py:114: TextXRegistrationError
python3.9-textx> _______________________________ test_subcommand ________________________________
python3.9-textx>     def test_subcommand():
python3.9-textx>         """
python3.9-textx>         Test that a command from the example project is registered.
python3.9-textx>         """
python3.9-textx> >       output = subprocess.check_output(['textx'], stderr=subprocess.STDOUT)
python3.9-textx> tests/functional/subcommands/test_subcommands.py:8:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:424: in check_output
python3.9-textx>     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:505: in run
python3.9-textx>     with Popen(*popenargs, **kwargs) as process:
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:951: in __init__
python3.9-textx>     self._execute_child(args, executable, preexec_fn, close_fds,
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> self = <Popen: returncode: 255 args: ['textx']>, args = ['textx']
python3.9-textx> executable = b'textx', preexec_fn = None, close_fds = True, pass_fds = ()
python3.9-textx> cwd = None, env = None, startupinfo = None, creationflags = 0, shell = False
python3.9-textx> p2cread = -1, p2cwrite = -1, c2pread = 11, c2pwrite = 12, errread = -1
python3.9-textx> errwrite = 12, restore_signals = True, gid = None, gids = None, uid = None
python3.9-textx> umask = -1, start_new_session = False
python3.9-textx>     def _execute_child(self, args, executable, preexec_fn, close_fds,
python3.9-textx>                        pass_fds, cwd, env,
python3.9-textx>                        startupinfo, creationflags, shell,
python3.9-textx>                        p2cread, p2cwrite,
python3.9-textx>                        c2pread, c2pwrite,
python3.9-textx>                        errread, errwrite,
python3.9-textx>                        restore_signals,
python3.9-textx>                        gid, gids, uid, umask,
python3.9-textx>                        start_new_session):
python3.9-textx>         """Execute program (POSIX version)"""
python3.9-textx>         if isinstance(args, (str, bytes)):
python3.9-textx>             args = [args]
python3.9-textx>         elif isinstance(args, os.PathLike):
python3.9-textx>             if shell:
python3.9-textx>                 raise TypeError('path-like args is not allowed when '
python3.9-textx>                                 'shell is true')
python3.9-textx>             args = [args]
python3.9-textx>         else:
python3.9-textx>             args = list(args)
python3.9-textx>         if shell:
python3.9-textx>             # On Android the default shell is at '/system/bin/sh'.
python3.9-textx>             unix_shell = ('/system/bin/sh' if
python3.9-textx>                       hasattr(sys, 'getandroidapilevel') else '/nix/store/ld19k12mf0pa42ff43qs3sizamp09f03-bash-5.1-p16/bin/sh')
python3.9-textx>             args = [unix_shell, "-c"] + args
python3.9-textx>             if executable:
python3.9-textx>                 args[0] = executable
python3.9-textx>         if executable is None:
python3.9-textx>             executable = args[0]
python3.9-textx>         sys.audit("subprocess.Popen", executable, args, cwd, env)
python3.9-textx>         if (_USE_POSIX_SPAWN
python3.9-textx>                 and os.path.dirname(executable)
python3.9-textx>                 and preexec_fn is None
python3.9-textx>                 and not close_fds
python3.9-textx>                 and not pass_fds
python3.9-textx>                 and cwd is None
python3.9-textx>                 and (p2cread == -1 or p2cread > 2)
python3.9-textx>                 and (c2pwrite == -1 or c2pwrite > 2)
python3.9-textx>                 and (errwrite == -1 or errwrite > 2)
python3.9-textx>                 and not start_new_session
python3.9-textx>                 and gid is None
python3.9-textx>                 and gids is None
python3.9-textx>                 and uid is None
python3.9-textx>                 and umask < 0):
python3.9-textx>             self._posix_spawn(args, executable, env, restore_signals,
python3.9-textx>                               p2cread, p2cwrite,
python3.9-textx>                               c2pread, c2pwrite,
python3.9-textx>                               errread, errwrite)
python3.9-textx>             return
python3.9-textx>         orig_executable = executable
python3.9-textx>         # For transferring possible exec failure from child to parent.
python3.9-textx>         # Data format: "exception name:hex errno:description"
python3.9-textx>         # Pickle is not used; it is complex and involves memory allocation.
python3.9-textx>         errpipe_read, errpipe_write = os.pipe()
python3.9-textx>         # errpipe_write must not be in the standard io 0, 1, or 2 fd range.
python3.9-textx>         low_fds_to_close = []
python3.9-textx>         while errpipe_write < 3:
python3.9-textx>             low_fds_to_close.append(errpipe_write)
python3.9-textx>             errpipe_write = os.dup(errpipe_write)
python3.9-textx>         for low_fd in low_fds_to_close:
python3.9-textx>             os.close(low_fd)
python3.9-textx>         try:
python3.9-textx>             try:
python3.9-textx>                 # We must avoid complex work that could involve
python3.9-textx>                 # malloc or free in the child process to avoid
python3.9-textx>                 # potential deadlocks, thus we do all this here.
python3.9-textx>                 # and pass it to fork_exec()
python3.9-textx>                 if env is not None:
python3.9-textx>                     env_list = []
python3.9-textx>                     for k, v in env.items():
python3.9-textx>                         k = os.fsencode(k)
python3.9-textx>                         if b'=' in k:
python3.9-textx>                             raise ValueError("illegal environment variable name")
python3.9-textx>                         env_list.append(k + b'=' + os.fsencode(v))
python3.9-textx>                 else:
python3.9-textx>                     env_list = None  # Use execv instead of execve.
python3.9-textx>                 executable = os.fsencode(executable)
python3.9-textx>                 if os.path.dirname(executable):
python3.9-textx>                     executable_list = (executable,)
python3.9-textx>                 else:
python3.9-textx>                     # This matches the behavior of os._execvpe().
python3.9-textx>                     executable_list = tuple(
python3.9-textx>                         os.path.join(os.fsencode(dir), executable)
python3.9-textx>                         for dir in os.get_exec_path(env))
python3.9-textx>                 fds_to_keep = set(pass_fds)
python3.9-textx>                 fds_to_keep.add(errpipe_write)
python3.9-textx>                 self.pid = _posixsubprocess.fork_exec(
python3.9-textx>                         args, executable_list,
python3.9-textx>                         close_fds, tuple(sorted(map(int, fds_to_keep))),
python3.9-textx>                         cwd, env_list,
python3.9-textx>                         p2cread, p2cwrite, c2pread, c2pwrite,
python3.9-textx>                         errread, errwrite,
python3.9-textx>                         errpipe_read, errpipe_write,
python3.9-textx>                         restore_signals, start_new_session,
python3.9-textx>                         gid, gids, uid, umask,
python3.9-textx>                         preexec_fn)
python3.9-textx>                 self._child_created = True
python3.9-textx>             finally:
python3.9-textx>                 # be sure the FD is closed no matter what
python3.9-textx>                 os.close(errpipe_write)
python3.9-textx>             self._close_pipe_fds(p2cread, p2cwrite,
python3.9-textx>                                  c2pread, c2pwrite,
python3.9-textx>                                  errread, errwrite)
python3.9-textx>             # Wait for exec to fail or succeed; possibly raising an
python3.9-textx>             # exception (limited in size)
python3.9-textx>             errpipe_data = bytearray()
python3.9-textx>             while True:
python3.9-textx>                 part = os.read(errpipe_read, 50000)
python3.9-textx>                 errpipe_data += part
python3.9-textx>                 if not part or len(errpipe_data) > 50000:
python3.9-textx>                     break
python3.9-textx>         finally:
python3.9-textx>             # be sure the FD is closed no matter what
python3.9-textx>             os.close(errpipe_read)
python3.9-textx>         if errpipe_data:
python3.9-textx>             try:
python3.9-textx>                 pid, sts = os.waitpid(self.pid, 0)
python3.9-textx>                 if pid == self.pid:
python3.9-textx>                     self._handle_exitstatus(sts)
python3.9-textx>                 else:
python3.9-textx>                     self.returncode = sys.maxsize
python3.9-textx>             except ChildProcessError:
python3.9-textx>                 pass
python3.9-textx>             try:
python3.9-textx>                 exception_name, hex_errno, err_msg = (
python3.9-textx>                         errpipe_data.split(b':', 2))
python3.9-textx>                 # The encoding here should match the encoding
python3.9-textx>                 # written in by the subprocess implementations
python3.9-textx>                 # like _posixsubprocess
python3.9-textx>                 err_msg = err_msg.decode()
python3.9-textx>             except ValueError:
python3.9-textx>                 exception_name = b'SubprocessError'
python3.9-textx>                 hex_errno = b'0'
python3.9-textx>                 err_msg = 'Bad exception data from child: {!r}'.format(
python3.9-textx>                               bytes(errpipe_data))
python3.9-textx>             child_exception_type = getattr(
python3.9-textx>                     builtins, exception_name.decode('ascii'),
python3.9-textx>                     SubprocessError)
python3.9-textx>             if issubclass(child_exception_type, OSError) and hex_errno:
python3.9-textx>                 errno_num = int(hex_errno, 16)
python3.9-textx>                 child_exec_never_called = (err_msg == "noexec")
python3.9-textx>                 if child_exec_never_called:
python3.9-textx>                     err_msg = ""
python3.9-textx>                     # The error must be from chdir(cwd).
python3.9-textx>                     err_filename = cwd
python3.9-textx>                 else:
python3.9-textx>                     err_filename = orig_executable
python3.9-textx>                 if errno_num != 0:
python3.9-textx>                     err_msg = os.strerror(errno_num)
python3.9-textx> >               raise child_exception_type(errno_num, err_msg, err_filename)
python3.9-textx> E               FileNotFoundError: [Errno 2] No such file or directory: 'textx'
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:1821: FileNotFoundError
python3.9-textx> ____________________________ test_subcommand_group _____________________________
python3.9-textx>     def test_subcommand_group():
python3.9-textx>         """
python3.9-textx>         Test that a command group is registered.
python3.9-textx>         """
python3.9-textx> >       output = subprocess.check_output(['textx', 'testgroup'],
python3.9-textx>                                          stderr=subprocess.STDOUT)
python3.9-textx> tests/functional/subcommands/test_subcommands.py:16:
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:424: in check_output
python3.9-textx>     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:505: in run
python3.9-textx>     with Popen(*popenargs, **kwargs) as process:
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:951: in __init__
python3.9-textx>     self._execute_child(args, executable, preexec_fn, close_fds,
python3.9-textx> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python3.9-textx> self = <Popen: returncode: 255 args: ['textx', 'testgroup']>
python3.9-textx> args = ['textx', 'testgroup'], executable = b'textx', preexec_fn = None
python3.9-textx> close_fds = True, pass_fds = (), cwd = None, env = None, startupinfo = None
python3.9-textx> creationflags = 0, shell = False, p2cread = -1, p2cwrite = -1, c2pread = 11
python3.9-textx> c2pwrite = 12, errread = -1, errwrite = 12, restore_signals = True, gid = None
python3.9-textx> gids = None, uid = None, umask = -1, start_new_session = False
python3.9-textx>     def _execute_child(self, args, executable, preexec_fn, close_fds,
python3.9-textx>                        pass_fds, cwd, env,
python3.9-textx>                        startupinfo, creationflags, shell,
python3.9-textx>                        p2cread, p2cwrite,
python3.9-textx>                        c2pread, c2pwrite,
python3.9-textx>                        errread, errwrite,
python3.9-textx>                        restore_signals,
python3.9-textx>                        gid, gids, uid, umask,
python3.9-textx>                        start_new_session):
python3.9-textx>         """Execute program (POSIX version)"""
python3.9-textx>         if isinstance(args, (str, bytes)):
python3.9-textx>             args = [args]
python3.9-textx>         elif isinstance(args, os.PathLike):
python3.9-textx>             if shell:
python3.9-textx>                 raise TypeError('path-like args is not allowed when '
python3.9-textx>                                 'shell is true')
python3.9-textx>             args = [args]
python3.9-textx>         else:
python3.9-textx>             args = list(args)
python3.9-textx>         if shell:
python3.9-textx>             # On Android the default shell is at '/system/bin/sh'.
python3.9-textx>             unix_shell = ('/system/bin/sh' if
python3.9-textx>                       hasattr(sys, 'getandroidapilevel') else '/nix/store/ld19k12mf0pa42ff43qs3sizamp09f03-bash-5.1-p16/bin/sh')
python3.9-textx>             args = [unix_shell, "-c"] + args
python3.9-textx>             if executable:
python3.9-textx>                 args[0] = executable
python3.9-textx>         if executable is None:
python3.9-textx>             executable = args[0]
python3.9-textx>         sys.audit("subprocess.Popen", executable, args, cwd, env)
python3.9-textx>         if (_USE_POSIX_SPAWN
python3.9-textx>                 and os.path.dirname(executable)
python3.9-textx>                 and preexec_fn is None
python3.9-textx>                 and not close_fds
python3.9-textx>                 and not pass_fds
python3.9-textx>                 and cwd is None
python3.9-textx>                 and (p2cread == -1 or p2cread > 2)
python3.9-textx>                 and (c2pwrite == -1 or c2pwrite > 2)
python3.9-textx>                 and (errwrite == -1 or errwrite > 2)
python3.9-textx>                 and not start_new_session
python3.9-textx>                 and gid is None
python3.9-textx>                 and gids is None
python3.9-textx>                 and uid is None
python3.9-textx>                 and umask < 0):
python3.9-textx>             self._posix_spawn(args, executable, env, restore_signals,
python3.9-textx>                               p2cread, p2cwrite,
python3.9-textx>                               c2pread, c2pwrite,
python3.9-textx>                               errread, errwrite)
python3.9-textx>             return
python3.9-textx>         orig_executable = executable
python3.9-textx>         # For transferring possible exec failure from child to parent.
python3.9-textx>         # Data format: "exception name:hex errno:description"
python3.9-textx>         # Pickle is not used; it is complex and involves memory allocation.
python3.9-textx>         errpipe_read, errpipe_write = os.pipe()
python3.9-textx>         # errpipe_write must not be in the standard io 0, 1, or 2 fd range.
python3.9-textx>         low_fds_to_close = []
python3.9-textx>         while errpipe_write < 3:
python3.9-textx>             low_fds_to_close.append(errpipe_write)
python3.9-textx>             errpipe_write = os.dup(errpipe_write)
python3.9-textx>         for low_fd in low_fds_to_close:
python3.9-textx>             os.close(low_fd)
python3.9-textx>         try:
python3.9-textx>             try:
python3.9-textx>                 # We must avoid complex work that could involve
python3.9-textx>                 # malloc or free in the child process to avoid
python3.9-textx>                 # potential deadlocks, thus we do all this here.
python3.9-textx>                 # and pass it to fork_exec()
python3.9-textx>                 if env is not None:
python3.9-textx>                     env_list = []
python3.9-textx>                     for k, v in env.items():
python3.9-textx>                         k = os.fsencode(k)
python3.9-textx>                         if b'=' in k:
python3.9-textx>                             raise ValueError("illegal environment variable name")
python3.9-textx>                         env_list.append(k + b'=' + os.fsencode(v))
python3.9-textx>                 else:
python3.9-textx>                     env_list = None  # Use execv instead of execve.
python3.9-textx>                 executable = os.fsencode(executable)
python3.9-textx>                 if os.path.dirname(executable):
python3.9-textx>                     executable_list = (executable,)
python3.9-textx>                 else:
python3.9-textx>                     # This matches the behavior of os._execvpe().
python3.9-textx>                     executable_list = tuple(
python3.9-textx>                         os.path.join(os.fsencode(dir), executable)
python3.9-textx>                         for dir in os.get_exec_path(env))
python3.9-textx>                 fds_to_keep = set(pass_fds)
python3.9-textx>                 fds_to_keep.add(errpipe_write)
python3.9-textx>                 self.pid = _posixsubprocess.fork_exec(
python3.9-textx>                         args, executable_list,
python3.9-textx>                         close_fds, tuple(sorted(map(int, fds_to_keep))),
python3.9-textx>                         cwd, env_list,
python3.9-textx>                         p2cread, p2cwrite, c2pread, c2pwrite,
python3.9-textx>                         errread, errwrite,
python3.9-textx>                         errpipe_read, errpipe_write,
python3.9-textx>                         restore_signals, start_new_session,
python3.9-textx>                         gid, gids, uid, umask,
python3.9-textx>                         preexec_fn)
python3.9-textx>                 self._child_created = True
python3.9-textx>             finally:
python3.9-textx>                 # be sure the FD is closed no matter what
python3.9-textx>                 os.close(errpipe_write)
python3.9-textx>             self._close_pipe_fds(p2cread, p2cwrite,
python3.9-textx>                                  c2pread, c2pwrite,
python3.9-textx>                                  errread, errwrite)
python3.9-textx>             # Wait for exec to fail or succeed; possibly raising an
python3.9-textx>             # exception (limited in size)
python3.9-textx>             errpipe_data = bytearray()
python3.9-textx>             while True:
python3.9-textx>                 part = os.read(errpipe_read, 50000)
python3.9-textx>                 errpipe_data += part
python3.9-textx>                 if not part or len(errpipe_data) > 50000:
python3.9-textx>                     break
python3.9-textx>         finally:
python3.9-textx>             # be sure the FD is closed no matter what
python3.9-textx>             os.close(errpipe_read)
python3.9-textx>         if errpipe_data:
python3.9-textx>             try:
python3.9-textx>                 pid, sts = os.waitpid(self.pid, 0)
python3.9-textx>                 if pid == self.pid:
python3.9-textx>                     self._handle_exitstatus(sts)
python3.9-textx>                 else:
python3.9-textx>                     self.returncode = sys.maxsize
python3.9-textx>             except ChildProcessError:
python3.9-textx>                 pass
python3.9-textx>             try:
python3.9-textx>                 exception_name, hex_errno, err_msg = (
python3.9-textx>                         errpipe_data.split(b':', 2))
python3.9-textx>                 # The encoding here should match the encoding
python3.9-textx>                 # written in by the subprocess implementations
python3.9-textx>                 # like _posixsubprocess
python3.9-textx>                 err_msg = err_msg.decode()
python3.9-textx>             except ValueError:
python3.9-textx>                 exception_name = b'SubprocessError'
python3.9-textx>                 hex_errno = b'0'
python3.9-textx>                 err_msg = 'Bad exception data from child: {!r}'.format(
python3.9-textx>                               bytes(errpipe_data))
python3.9-textx>             child_exception_type = getattr(
python3.9-textx>                     builtins, exception_name.decode('ascii'),
python3.9-textx>                     SubprocessError)
python3.9-textx>             if issubclass(child_exception_type, OSError) and hex_errno:
python3.9-textx>                 errno_num = int(hex_errno, 16)
python3.9-textx>                 child_exec_never_called = (err_msg == "noexec")
python3.9-textx>                 if child_exec_never_called:
python3.9-textx>                     err_msg = ""
python3.9-textx>                     # The error must be from chdir(cwd).
python3.9-textx>                     err_filename = cwd
python3.9-textx>                 else:
python3.9-textx>                     err_filename = orig_executable
python3.9-textx>                 if errno_num != 0:
python3.9-textx>                     err_msg = os.strerror(errno_num)
python3.9-textx> >               raise child_exception_type(errno_num, err_msg, err_filename)
python3.9-textx> E               FileNotFoundError: [Errno 2] No such file or directory: 'textx'
python3.9-textx> /nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/subprocess.py:1821: FileNotFoundError
python3.9-textx> =============================== warnings summary ===============================
python3.9-textx> tests/functional/test_examples.py:14
python3.9-textx>   /build/source/tests/functional/test_examples.py:14: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
python3.9-textx>     import imp
python3.9-textx> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
python3.9-textx> =========================== short test summary info ============================
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_object_processor_with_optional_parameter_default
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_object_processor_with_optional_parameter_on
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_object_processor_with_optional_parameter_off
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_check_metamodel - as...
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_check_valid_model - ...
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_check_valid_model_with_explicit_language_name
python3.9-textx> FAILED tests/functional/registration/test_check.py::test_check_invalid_model
python3.9-textx> FAILED tests/functional/registration/test_discovery.py::test_list_languages_cli
python3.9-textx> FAILED tests/functional/registration/test_discovery.py::test_list_generators_cli
python3.9-textx> FAILED tests/functional/registration/test_generate.py::test_generator_registered
python3.9-textx> FAILED tests/functional/registration/test_generate.py::test_generating_flow_model
python3.9-textx> FAILED tests/functional/registration/test_generate.py::test_generate_by_providing_explicit_language_name
python3.9-textx> FAILED tests/functional/registration/test_generate.py::test_passing_custom_arguments_to_generator
python3.9-textx> FAILED tests/functional/registration/test_registration.py::test_declaratively_registered_generator_always_available
python3.9-textx> FAILED tests/functional/registration/projects/data_dsl/tests/test_data_dsl.py::test_data_dsl
python3.9-textx> FAILED tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py::test_flow_dsl
python3.9-textx> FAILED tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py::test_flow_dsl_validation
python3.9-textx> FAILED tests/functional/registration/projects/flow_dsl/tests/test_flow_dsl.py::test_flow_dsl_types_validation
python3.9-textx> FAILED tests/functional/registration/projects/flow_dsl/tests/test_issue196.py::test_issue196_errors_in_scope_provider_and_obj_processor
python3.9-textx> FAILED tests/functional/registration/projects/flow_dsl/tests/test_issue196.py::test_issue196_syntax_error_1
python3.9-textx> FAILED tests/functional/registration/projects/flow_dsl/tests/test_issue196.py::test_issue196_syntax_error_2
python3.9-textx> FAILED tests/functional/registration/projects/types_dsl/tests/test_types_dsl.py::test_types_dsl
python3.9-textx> FAILED tests/functional/registration/projects/types_dsl/tests/test_types_dsl.py::test_types_dsl_invalid
python3.9-textx> FAILED tests/functional/subcommands/test_subcommands.py::test_subcommand - Fi...
python3.9-textx> FAILED tests/functional/subcommands/test_subcommands.py::test_subcommand_group
python3.9-textx> ================== 25 failed, 306 passed, 1 warning in 10.82s ==================
```